swift/IceStorm/clock/Sources/Publisher/main.swift

```diff
@@ -18,114 +18,106 @@ enum Option: String {
     case oneway = "--oneway"
 }
 
-func run(ctrlCHandler: Ice.CtrlCHandler) async -> Int32 {
-    do {
-        var args = [String](CommandLine.arguments.dropFirst())
+// CtrlCHandler is a helper class that handles Ctrl+C and similar signals. It must be constructed at the beginning
+// of the program, before creating an Ice communicator or starting any thread.
+let ctrlCHandler = Ice.CtrlCHandler()
 
-        let communicator = try Ice.initialize(args: &args, configFile: "config.pub")
-        defer {
-            communicator.destroy()
-        }
+// Create an Ice communicator to initialize the Ice runtime.
+var args = [String](CommandLine.arguments.dropFirst())
+let communicator = try Ice.initialize(args: &args, configFile: "config.pub")
+defer {
+    communicator.destroy()
+}
 
-        var option: Option = .none
-        var topicName = "time"
-
-        for var i in 0 ..< args.count {
-            let oldoption = option
-
-            if args[i].starts(with: "--") {
-                guard let o = Option(rawValue: args[i]) else {
-                    usage()
-                    return 1
-                }
-                option = o
-            } else {
-                topicName = args[i]
-                i += 1
-                break
-            }
-
-            if oldoption != option, oldoption != .none {
-                usage()
-                return 1
-            }
-        }
+// Parse command-line options.
 
-        guard let base = try communicator.propertyToProxy("TopicManager.Proxy"),
-            let manager = try await checkedCast(prx: base, type: IceStorm.TopicManagerPrx.self) else {
-            print("invalid proxy")
-            return 1
-        }
+var option: Option = .none
+var topicName = "time"
 
-        //
-        // Retrieve the topic.
-        //
-        var topic: IceStorm.TopicPrx!
-        do {
-            topic = try await manager.retrieve(topicName)
-        } catch is IceStorm.NoSuchTopic {
-            do {
-                topic = try await manager.create(topicName)
-            } catch is IceStorm.TopicExists {
-                print("temporary error. try again.")
-                return 1
-            }
-        }
+for var i in 0..<args.count {
+    let oldOption = option
```

Comment:
Can't we get rid of `oldOption` and  also remove 

```
if oldOption != option, oldOption != .none {
        usage()
        exit(1)
    }
```

We can check this before doing `option = o`

Reply:
I didn't change this code (except oldoption => oldOption) ... a lot more demo code to cleanup.